eCommerce platforms/Adobe Magento/src/code/local/Microsoft/Dfp/controllers/Checkout/OnepageController.php

```diff
@@ -0,0 +1,399 @@
+					'expirationDate'  => $expirationDateISO,
+					'lastFourDigits'  => $PaymentData['cc_last4'],
+					'purchaseAmount'  => $order->getGrandTotal(),
+					'billingAddress'  => $billingAddress
+				);
+				$paymentInstrumentList[] = $paymentInstrument;
+				break;
+			case 'checkmo':
+				$paymentInstrument = array(
+					'type'			 => 'Invoice',
+					'purchaseAmount' => $order->getGrandTotal(),
+					'billingAddress' => $billingAddress
+				);
+				$paymentInstrumentList[] = $paymentInstrument;
+				break;
+		}
+		return $paymentInstrumentList;
+	}
+
+	private function generatePurchasePayload($order, $assessmentType)
+	{
+		$billingAddress = $order->getBillingAddress();
+
+		$quote = Mage::getModel('sales/quote')->load($order->getQuoteId());
+		$userId = $order->customer_email;
+		if ($quote->getCheckoutMethod(true) == 'guest') {
+			$userId = $this->dfp->GUID();
+		}
+
+		// User Details
+		$userCountryCode = 'USA'; //this is something needs to set based on zip code, here we just hardcode a possible one
+		$userCountryDialCode = '1'; //this is something needs to set based on zip code, here we just hardcode a possible one
+		$timeZone = (Mage::getSingleton('core/session')->getCustomerTimeZone()) / 60;
+		$timeZone = ($timeZone > 0 ? '-' : '+') . $timeZone;
+		$user = array(
+			'userId'					=> $userId,
+			'firstName'					=> $order->customer_firstname,
+			'lastName' 					=> $order->customer_lastname,
+			'country'					=> $userCountryCode,
+			'zipCode'					=> $billingAddress->postcode,
+			"timeZone"					=> $timeZone,
+			'language'					=> strtoupper(substr(Mage::app()->getLocale()->getLocaleCode(), 0, 2)),
+			'phoneNumber'				=> "+" . $userCountryDialCode . "-" . $billingAddress->telephone,
+			'email'						=> $order->customer_email,
+			'isEmailValidated'			=> false,
+			'isPhoneNumberValidated'	=> false
+		);
+
+		// Shipping Address Details
+		$shippingAddress = $order->getShippingAddress();
+		$shipStreet = $shippingAddress->getStreet();
+		if ($shippingAddress)
+			$shippingData = $shippingAddress->getData();
+
+		$shippingCountryCode = 'USA';
+		$shippingCountryDialCode = '1';
+
+		$orderShippingAddress = array(
+			'firstName'		=> $shippingData['firstname'],
+			'lastName'		=> $shippingData['lastname'],
+			'phoneNumber'	=> "+" . $shippingCountryDialCode . "-" . $shippingData['telephone'],
+			'street1'		=> $shipStreet[0],
+			'street2'		=> count($shipStreet) > 1 ? $shipStreet[1] : null,
+			'city'			=> $shippingAddress->city,
+			'state'			=> $shippingAddress->region,
+			'zipCode'		=> $shippingAddress->postcode,
+			'country'		=> $shippingCountryCode
+		);
+
+		// Device FingerPrinting Details
+		$deviceContext = array(
+			"deviceContextId"	=> Mage::getSingleton('core/session')->getFptDfpSessionId() ?: $this->dfp->GUID(),
+			"ipAddress"			=> getenv('HTTP_CLIENT_IP') ?: getenv('HTTP_X_FORWARDED_FOR') ?: getenv('HTTP_X_FORWARDED') ?: getenv('HTTP_FORWARDED_FOR') ?: getenv('HTTP_FORWARDED') ?: getenv('REMOTE_ADDR'),
```

Comment:
Same comment here about ??

Reply:
Fixed across the board anywhere ?: was used.